plugins/out_stackdriver/stackdriver.c

```diff
@@ -1132,14 +1176,21 @@ static int stackdriver_format(struct flb_config *config,
                                 &operation_first, &operation_last, &mp_pck);
         }
 
+        /* labels */
+        if (labels_ptr != NULL) {
+            msgpack_pack_str(&mp_pck, 6);
+            msgpack_pack_str_body(&mp_pck, "labels", 6);
+            msgpack_pack_object(&mp_pck, *labels_ptr);
```

Comment:
If the value of the custom labels is a nested struct, it would be fully converted?

Reply:
I will say yes because **msgpack_pack_object** is using here to pack object. A good example here will be that if *labels_ptr points to the value field of a map and lets say that the map is [A: {B:b, C:c}] then {B:b, C:c} will be packed under "labels". Is that what we need here?

Reply:
I meant the conversion as {labels_key: {A: {B:b}}} -> {"labels": {A: {B, b}}. The value of labels_ptr is nested. Even though I am not sure whether we would have nested struct in real use cases.

Reply:
This is convert the value of labels_key (no matter what types it is) to the value of "labels". This also applies for the case that the value of labels_ptr is nested. But I doubt that this will be rejected by stackdriver as it expects map for the value of "labels"

Reply:
Note: Do local testing on for nested object of labels_ptr